addons/pos_self_order/models/ir_http.py

```diff
@@ -19,8 +19,21 @@ def _get_translation_frontend_modules_name(cls):
     # This override works around the issue.
     @api.model
     def get_nearest_lang(self, lang_code: str) -> str:
-        if lang_code and '/pos-self/' in request.httprequest.path:
-            config_id_match = re.search(r'/pos-self/(\d+)', request.httprequest.path)
+        if not lang_code:
+            return super().get_nearest_lang(lang_code)
+
+        referer_url = request.httprequest.headers.get('Referer', '')
+        path = request.httprequest.path
+
+        if '/pos-self/' in path:
+            path_with_config = path
+        elif '/pos-self/' in referer_url:
+            path_with_config = referer_url
+        else:
+            path_with_config = None
```

Comment:
did not test it, but that's what I have in mind
```suggestion
        if path.startswith('/pos-self/'):
            path_with_config = path
        elif referer_url.startswith('/pos-self/') and path == '/website/translations':
            path_with_config = referer_url
        else:
            path_with_config = None
```

Reply:
@ushyme mmmmh maybe you are right

Reply:
@Julien00859 Yep, I did some testing since then, and it does break when using `startswith`.
Of course, we could do some more complicated parsing based on whether the first path segment is (close to) a valid language, but I don't know if it's worth it in this case?

Reply:
Done!